libkbfs/block_journal.go

```diff
@@ -495,65 +498,30 @@ func (j *blockJournal) addReference(
 		}
 	}()
 
-	if j.isShutdown {
-		return errBlockJournalShutdown
-	}
-
-	refs := j.refs[id]
-	if refs == nil {
-		return BServerErrorBlockNonExistent{fmt.Sprintf("Block ID %s "+
-			"doesn't exist and cannot be referenced.", id)}
-	}
-
-	// Only add it if there's a non-archived reference.
-	hasNonArchivedRef := false
-	for _, refEntry := range refs {
-		if refEntry.Status == liveBlockRef {
-			hasNonArchivedRef = true
-			break
-		}
-	}
-
-	if !hasNonArchivedRef {
-		return BServerErrorBlockArchived{fmt.Sprintf("Block ID %s has "+
-			"been archived and cannot be referenced.", id)}
-	}
-
-	// TODO: Figure out if we should allow adding a reference even
-	// if all the existing references are archived, or if we have
-	// no references at all. Also figure out what to do with an
-	// addReference without a preceding Put.
-
-	err = j.putRefEntry(id, blockRefEntry{
-		Status:  liveBlockRef,
-		Context: context,
-	})
+	ordinal, err := j.appendJournalEntry(addRefOp,
```

Comment:
I actually just moved the check to `BlockServerDisk.ArchiveBlockReferences`, since it's the only thing that needs the check. So now `blockJournal` is more like a dumb data store. Does that clarify things?